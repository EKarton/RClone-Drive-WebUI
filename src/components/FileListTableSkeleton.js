import Paper from '@mui/material/Paper';
import Skeleton from '@mui/material/Skeleton';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import PropTypes from 'prop-types';
import './FileListTableSkeleton.scss';

export default function FileListTableSkeleton({ numRows }) {
  return (
    <TableContainer
      component={Paper}
      data-testid="files-list-table-skeleton"
      className="file-list-table-skeleton"
    >
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>Date Modified</TableCell>
            <TableCell>File Size</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {Array.from({ length: numRows }, (_, i) => (
            <TableRow key={i} className="file-list-table-skeleton__row">
              <TableCell colSpan={3}>
                <Skeleton animation="wave" />
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

FileListTableSkeleton.defaultProps = {
  numRows: 5,
};

FileListTableSkeleton.propType = {
  numRows: PropTypes.number,
};
