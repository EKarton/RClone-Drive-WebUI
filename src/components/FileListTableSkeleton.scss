.file-list-table-skeleton {
  margin-bottom: 0 !important;
  box-shadow: none !important;
  border: 1px solid rgba(34 36 38 / 15%);

  &__row {
    &:last-child td {
      border: 0;
    }
  }
}
