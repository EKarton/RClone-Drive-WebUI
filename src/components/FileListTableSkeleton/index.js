import Skeleton from '@mui/material/Skeleton';
import TableCell from '@mui/material/TableCell';
import TableRow from '@mui/material/TableRow';
import PropTypes from 'prop-types';
import FileListTable from '../FileListTable';
import './index.scss';

export default function FileListTableSkeleton({ numRows }) {
  return (
    <FileListTable data-testid="files-list-table-skeleton">
      {Array.from({ length: numRows }, (_, i) => (
        <TableRow key={i} className="file-list-table-skeleton__row">
          <TableCell colSpan={3}>
            <Skeleton animation="wave" />
          </TableCell>
        </TableRow>
      ))}
    </FileListTable>
  );
}

FileListTableSkeleton.defaultProps = {
  numRows: 5,
};

FileListTableSkeleton.propTypes = {
  numRows: PropTypes.number,
};
