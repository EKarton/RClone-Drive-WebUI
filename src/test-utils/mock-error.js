/**
 * Replaces the error object's stack trace with mock data
 * @param {Error} error the Error object
 * @returns {Error} the same error, but with the stack trace mocked
 */
export function mockErrorStackTrace(error) {
  error.stack = 'Mock content for stack trace';
  return error;
}
