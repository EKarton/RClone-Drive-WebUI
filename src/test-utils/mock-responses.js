export const mockFiles = {
  list: [
    {
      Path: '.rclone',
      Name: '.rclone',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-11-01T21:09:05.237Z',
      IsDir: true,
      ID: '1a4mI4vsGKypqFFPwLnrFid1ZdvlhirDi',
    },
    {
      Path: 'Documents',
      Name: 'Documents',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T01:51:05.982Z',
      IsDir: true,
      ID: '1_za5Kqy9ZjXZ7rsztGpKhtJLTuNCKv8q',
    },
    {
      Path: 'Games',
      Name: 'Games',
      Size: -9,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T05:14:59.237Z',
      IsDir: true,
      ID: '1j6mAjxQoJokiU8yWJYgyWYIiSIgH1G7M',
    },
    {
      Path: 'Literature',
      Name: 'Literature',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-29T07:10:05.059Z',
      IsDir: true,
      ID: '1dj1T5YkjRQZNiQ5efCA8PzideJb88d3K',
    },
    {
      Path: 'Pictures',
      Name: 'Pictures',
      Size: -10,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:39:11.850Z',
      IsDir: true,
      ID: '1D2PpmoIT23NqyTnolmohZ0UxvOd2D7n4',
    },
    {
      Path: 'Software',
      Name: 'Software',
      Size: -2,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T03:45:02.471Z',
      IsDir: true,
      ID: '1zzr0Bwp9tel66rCUKz8mZIXRBR2kuywr',
    },
    {
      Path: 'Videos',
      Name: 'Videos',
      Size: -8,
      MimeType: 'inode/directory',
      ModTime: '2021-09-28T04:22:24.997Z',
      IsDir: true,
      ID: '1eJlJ1PT7b9XY3Q5tLeU49BNkug259986',
    },
    {
      Path: 'backup.sh',
      Name: 'backup.sh',
      Size: 1110,
      MimeType: 'application/x-sh',
      ModTime: '2021-11-14T18:58:44.655Z',
      IsDir: false,
    },
  ],
};

export const mockRemotes = {
  remotes: ['googledrive', 'onedrive', 'icloud'],
};

export const mockPictures = {
  list: [
    {
      Path: 'Pictures/2012/20120517_171428.JPG',
      Name: '20120517_171428.JPG',
      Size: 335408,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:14:30.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_135206.JPG',
      Name: '20120517_135206.JPG',
      Size: 262719,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T17:52:06.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_204557.JPG',
      Name: '20120516_204557.JPG',
      Size: 245109,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T00:45:58.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_174510.JPG',
      Name: '20120517_174510.JPG',
      Size: 271914,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T21:45:12.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120820_144833.JPG',
      Name: '20120820_144833.JPG',
      Size: 2662880,
      MimeType: 'image/jpeg',
      ModTime: '2012-08-20T18:48:32.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120517_094409.JPG',
      Name: '20120517_094409.JPG',
      Size: 280596,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-17T13:44:10.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181954.JPG',
      Name: '20120516_181954.JPG',
      Size: 416424,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:19:54.000Z',
      IsDir: false,
    },
    {
      Path: 'Pictures/2012/20120516_181809.JPG',
      Name: '20120516_181809.JPG',
      Size: 211575,
      MimeType: 'image/jpeg',
      ModTime: '2012-05-16T22:18:10.000Z',
      IsDir: false,
    },
  ],
};
