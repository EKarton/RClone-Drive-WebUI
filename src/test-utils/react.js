import { FileViewerProvider } from 'contexts/FileViewerStore';
import { RCloneInfoStateProvider } from 'contexts/RCloneInfoStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState, initialRCloneInfoState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  const renderedComponent = render(
    <RCloneInfoStateProvider defaultState={initialRCloneInfoState}>
      <FileViewerProvider defaultState={initialFileViewerState}>
        <Router history={history}>{component}</Router>
      </FileViewerProvider>
    </RCloneInfoStateProvider>
  );

  return {
    ...renderedComponent,
    history,
  };
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
