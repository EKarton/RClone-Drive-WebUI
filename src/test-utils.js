import { FileViewerProvider, initialState } from 'contexts/FileViewerStore';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router';

const customRender = (
  component,
  { initialFileViewerState = initialState } = {},
  { route = '/' } = {}
) => {
  const history = createMemoryHistory({ initialEntries: [route] });

  return render(
    <FileViewerProvider defaultState={initialFileViewerState}>
      <Router history={history}>{component}</Router>
    </FileViewerProvider>
  );
};

export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { customRender, userEvent };
