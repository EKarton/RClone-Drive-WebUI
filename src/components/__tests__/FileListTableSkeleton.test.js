import { render } from 'test-utils';
import FileListTableSkeleton from '../FileListTableSkeleton';

describe('FileListTableSkeleton', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(<FileListTableSkeleton />);

    expect(baseElement).toMatchSnapshot();
  });
});
