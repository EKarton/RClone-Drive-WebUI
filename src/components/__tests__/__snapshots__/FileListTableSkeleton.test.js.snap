// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileListTableSkeleton should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root file-list-table css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      data-testid="files-list-table-skeleton"
    >
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium file-list-table__header-cell css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium file-list-table__header-cell css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Date Modified
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium file-list-table__header-cell css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              File Size
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root file-list-table__body css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root file-list-table-skeleton__row css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              colspan="3"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
              />
            </td>
          </tr>
          <tr
            class="MuiTableRow-root file-list-table-skeleton__row css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              colspan="3"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
              />
            </td>
          </tr>
          <tr
            class="MuiTableRow-root file-list-table-skeleton__row css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              colspan="3"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
              />
            </td>
          </tr>
          <tr
            class="MuiTableRow-root file-list-table-skeleton__row css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              colspan="3"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
              />
            </td>
          </tr>
          <tr
            class="MuiTableRow-root file-list-table-skeleton__row css-1q1u3t4-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              colspan="3"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;
